Add minecraft-quilt template

rickiewars opened the pull request:
Solves issue #143

LordRalex committed:
Move templates to new structure for v2.5

LordRalex committed:
Revert "Move templates to new structure for v2.5"

This reverts commit 68a5cc40c891fd64347f7c5427b00a8d24a44e42.

rickiewars committed:
Add minecraft-quilt template

rickiewars commented:
I've successfully been able to create a new quilt server inside my pufferpanel instance of Docker.
Quilt is an upcomming successor of Fabric and I'm seeing mods dropping support for Fabric in fafor of Quilt.

thecmdradama committed:
Change parameters to specify the config file and a specific world (#192)

LordRalex commented:
Please rebase this PR to v2.6 and follow our formatting rules:
https://github.com/PufferPanel/templates/blob/v2.6/README.md#formattng-rules

Please also confirm there is not a corrected template in the new branch.

rickiewars committed:
Merge branch 'PufferPanel:v2' into v2

rickiewars committed:
Merge remote-tracking branch 'upstream/v2.6' into v2

rickiewars committed:
Updated configuration to comply with formatting rules

rickiewars commented:
> Please rebase this PR to v2.6 and follow our formatting rules: https://github.com/PufferPanel/templates/blob/v2.6/README.md#formattng-rules
> 
> Please also confirm there is not a corrected template in the new branch.

I rebased the pr to v2.6 and updated the file to comply with the formatting rules and tested the template locally for good measure.
I also confirm that the template is not yet in the new branch.

I hope it can be merged soon.

LordRalex commented:
Your PR is targeting the v2 branch, which needs to be the v2.6 branch.

rickiewars commented:
How do I change this target branch? Or must I create a new pull-request for this?

LordRalex commented:
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request

rickiewars commented:
Changed the base to v2.6
Please ignore my created pull request #196

LordRalex commented on minecraft-quilt/minecraft-quilt.json:
```diff
@@ -0,0 +1,115 @@
+      "required": true,
+      "value": "0.0.0.0"
+    },
+    "java": {
```
As of v2.6, we use the javadl option to handle the java requirements.
See https://github.com/PufferPanel/templates/blob/v2.6/minecraft-vanilla/minecraft-vanilla.json#L52-L55 and https://github.com/PufferPanel/templates/blob/v2.6/minecraft-vanilla/minecraft-vanilla.json#L73 for example usages.

LordRalex commented on minecraft-quilt/minecraft-quilt.json:
```diff
@@ -0,0 +1,115 @@
+    },
+    {
+      "source": "quilt-installer.jar",
+      "target": "installer/quilt-installer.jar",
```
Any reason to keep the installer instead of deleting it?

rickiewars replied:
No can be deleted indeed. This file was based on the Fabric config and it was done over there previously. That's why I kept it in

rickiewars commented:
I'm not sure how terraria-tshock/terraria-tshock.json got edited, is that change correct?

LordRalex commented:
That file is from v2, and not v2.6's file. Not sure if it's good though or not. If you can revert the changes, it'd be better.

rickiewars committed:
Applied feedback from @LordRalex

rickiewars committed:
Reverted change from V2

rickiewars committed:
Changed Java variable to java17 command

rickiewars commented:
I reverted that file and applied your feedback

LordRalex commented on minecraft-quilt/minecraft-quilt.json:
```diff
@@ -0,0 +1,118 @@
+{
+  "name": "undefined",
```
Name is no longer defined correctly

LordRalex commented on minecraft-quilt/minecraft-quilt.json:
```diff
@@ -0,0 +1,118 @@
+    "post": [],
+    "environmentVars": {}
+  },
+  "data": {
```
Please put the data object after display, per https://github.com/PufferPanel/templates#formattng-rules

rickiewars committed:
Update minecraft-quilt.json

rickiewars commented:
Applied your feedback.

Conserning the formatting rules:

> I think everything was shuffled again whin I saved the config in Pufferfish for testing. Some lines needed changing so out of conveniance, I copied the working config directly out of PufferFish. I think there is an improvement here for Pufferfish to show the saved template to the user according to the formatting rules.
